### ISSUE TITLE
Add more properties to right-hand panel for `street` and `streetmix-loader` component

### ISSUE THREAD

**Issue by kfarr** [3DStreet/3dstreet-editor#14]
Now that we have the [right panel working](https://github.com/3DStreet/3dstreet-editor/pull/12) for global properties, the next step is to expose properties for specific components. The first to focus on would be for entities with `street` and/or `streetmix-loader` component.

The design reference is here:
https://www.figma.com/file/W9jid3A0jgssnIKBFEvG38/3DStreet-2022?node-id=896%3A16119

It's ok if other entities with other components do not display correctly, I am suggesting that we focus on one at a time to have a reasonable pace.

**Comment by Algorush** [comment]
I have questions about a new task for the right panel.
1. If an entity has a street component, does it also have a streetmix-loader component, or is it optional?
2. For the street-JSON field, where should the information come from?
3. What do two Refresh buttons?
4. The 'Edit in Streetmix' button opens a link from streetmix-URL?
5. In the index.html file for entities with streetmix-loader components, there is only "showBuildings: false;". In Figma design, there are also switches for "Show ground", "Road striping". If they are not in the component, then use the default value of False?

**Comment by kfarr** [comment]
Hi @Algorush thanks for these, all good questions.

I answer questions below, but to keep things simpler for a first commit I suggest:
* Only expose the values that come from the actual scheme, do not add additional buttons or widgets with custom logic yet
* We need to have separate sections for the components just like the default inspector. The design does not accommodate this yet, I will ask Daria to help.

> 1. If an entity has a street component, does it also have a streetmix-loader component, or is it optional?

It is optional -- there will be cases when an entity with a `street` component will NOT have a `streetmix-loader`. So likely you will need to change from the design in order to have the fields from each component separated, not mixed together.

> 2. For the street-JSON field, where should the information come from?

This is populated at runtime by the streetmix-loader, there should not be any additional changes needed for this to work.

> 3. What do two Refresh buttons?

The intent is that they reload the entity with the new property values. Let's wait on this.

> 4. The 'Edit in Streetmix' button opens a link from streetmix-URL?

Yes, but let's wait on this.

> 5. In the index.html file for entities with streetmix-loader components ...

Only expose properties defined in the component schema for now, do not add additional ones even if they're in the Figma.

**Comment by kfarr** [comment]
Question: Starting from the default inspector that exposes all properties of a component for editing, how to define which properties to display / hide / rename or change widget (to button or long text field?)

A few options:

1. Write custom logic in the editor that determines which fields to display depending on the component name. For example if `street` then display custom widgets as defined via custom logic inside of the 3dstreet editor.
2. Define at the entity level with an attribute such as `data-editor-hide-props="prop1, prop2, prop3"`
3. Define at the component level. There are a few options for this:
   - A. Create a new method on every component that we want to have custom UI for. This method returns an object that indicates which fields should be displayed, hidden, and if certain field should use a custom widget.
   - B. Use the property name to indicate which custom widget should be rendered. For example, if a `boolean` field is named `buttonFoo_Bar` render a button widget (instead of a checkbox) with the display text "Foo Bar"
   - C. Add extra values to the schema to indicate rendering preference. Such as below:

```
AFRAME.registerComponent('street', {
  schema: {
    JSON: { type: 'string', editorWidget: 'large-text-field' },
    type: { default: 'streetmixSegmentsFeet' }, 
    left: { default: '', editorDisplay: false },
    right: { default: '', editorDisplay: false },
    showGround: { default: true },
    showStriping: { default: true },
    length: { default: 150 },
    actionRefresh: { type: 'boolean', editorWidget: 'button', editorDisplayName: 'Refresh Street' }
  },
```

Suggested answer:
3C seems like the "right" answer in the long-term, but it might be easier to start with 1 and then later migrate to 3C.

@Algorush what do you think? Also posted to Stack Overflow to get some ideas: https://stackoverflow.com/questions/73271455/method-for-specifying-widget-behavior-for-component-properties-with-a-custom-a-f

**Comment by Algorush** [comment]
Yes, we can start with option 1 for now. For now, we can only display those fields/switches, etc. for which there is data in the component.
I also noticed on Figma that there is no X, Y, Z fields for selecting Scale for entities with 'street' component, right?
And there are also entities with 'street' components that have many other fields, like this one:
https://www.figma.com/file/W9jid3A0jgssnIKBFEvG38/3DStreet-2022?node-id=792%3A9506
Can you tell please, where does the data for these fields come from?

**Comment by kfarr** [comment]
https://github.com/3DStreet/3dstreet-editor/pull/22
